op-challenger/game/monitor.go

```diff
@@ -102,24 +110,29 @@ func (m *gameMonitor) progressGames(ctx context.Context, blockNum uint64) error
 func (m *gameMonitor) MonitorGames(ctx context.Context) error {
 	m.logger.Info("Monitoring fault dispute games")
 
-	blockNum := uint64(0)
+	onNewL1Head := func(ctx context.Context, sig eth.L1BlockRef) {
+		if err := m.progressGames(ctx, sig.Number); err != nil {
+			m.logger.Error("Failed to progress games", "err", err)
+		}
+	}
+
+	resubFn := func(innerCtx context.Context, err error) (event.Subscription, error) {
+		if ctx.Err() != nil {
+			return nil, ctx.Err()
+		}
+		if err != nil {
+			m.logger.Warn("resubscribing after failed L1 subscription", "err", err)
+		}
+		return eth.WatchHeadChanges(ctx, m.l1Source, onNewL1Head)
+	}
+
+	m.l1HeadsSub = event.ResubscribeErr(10*time.Second, resubFn)
 	for {
 		select {
 		case <-ctx.Done():
-			return ctx.Err()
-		default:
-			nextBlockNum, err := m.fetchBlockNumber(ctx)
-			if err != nil {
-				m.logger.Error("Failed to load current block number", "err", err)
-				continue
-			}
-			if nextBlockNum > blockNum {
-				blockNum = nextBlockNum
-				if err := m.progressGames(ctx, nextBlockNum); err != nil {
-					m.logger.Error("Failed to progress games", "err", err)
-				}
-			}
-			if err := m.clock.SleepCtx(ctx, time.Second); err != nil {
+			return nil
+		case err, ok := <-m.l1HeadsSub.Err():
+			if !ok {
 				return err
 			}
```

Comment:
I feel like this is going to silently ignore errors.  In the op-node version if `ok` it logs the error.

Reply:
Agree, added it back. Accidentally had removed it as part of a cleanup.